3scale/porta
# Fix application plan changes with different services in the API

**What this PR does / why we need it**

It prevents application plan changes between different services of the same provider through the API.

**Which issue(s) this PR fixes** 

Closes [THREESCALE-1869](https://issues.jboss.org/browse/THREESCALE-1869)

**Verification steps** 

As a provider with more than a one service containing application plans, use the 3scale API to change a buyer application to a plan that belongs to a service different than the current service the application is associated with through its current plan. For example,

- Provider `X` has services `S1` and `S2`
- `S1` issued application plan `P1`
- `S2` issued application plan `P1`
- Developer `Y` has an application `A` currently on plan `P1`
- Provider `X` performs an API PUT request to `/admin/api/accounts/<Y:id>/applications/<A:id>/change_plan.xml` with payload `{ plan_id: <P2:id> }`

The response should now be a 404, as `P2` does not belong to `S1`.
